src/main/java/org/jivesoftware/smackx/muc/MultiUserChatOwnerRevokeAdminIntegrationTest.java

```diff
@@ -0,0 +1,577 @@
+        }
+    }
+
+    /**
+     * Verifies that an owner can revoke admin status from a user that is currently a participant in the room.
+     */
+    @SmackIntegrationTest(section = "10.7", quote = "An owner might want to revoke a user's admin status; this is done by changing the user's affiliation to something other than \"admin\" or \"owner\"")
+    public void testRevokeAdminWhileInRoom() throws Exception
+    {
+        // Setup test fixture.
+        final EntityBareJid mucAddress = getRandomRoom("smack-inttest-owner-admin-revoke-inroom");
+        final MultiUserChat mucAsSeenByOwner = mucManagerOne.getMultiUserChat(mucAddress);
+        final MultiUserChat mucAsSeenByTarget = mucManagerTwo.getMultiUserChat(mucAddress);
+
+        final Resourcepart nicknameOwner = Resourcepart.from("owner-" + randomString);
+        final Resourcepart nicknameTarget = Resourcepart.from("target-" + randomString);
+
+        final EntityFullJid targetMucAddress = JidCreate.entityFullFrom(mucAddress, nicknameTarget);
+
+        createMuc(mucAsSeenByOwner, nicknameOwner);
+        try {
+            final SimpleResultSyncPoint ownerSeesTarget = new SimpleResultSyncPoint();
+            mucAsSeenByOwner.addParticipantStatusListener(new ParticipantStatusListener() {
+                @Override
+                public void joined(EntityFullJid participant) {
+                    if (participant.equals(targetMucAddress)) {
+                        ownerSeesTarget.signal();
+                    }
+                }
+            });
+            mucAsSeenByTarget.join(nicknameTarget);
+            ownerSeesTarget.waitForResult(timeout);
+
+            final MUCAdmin grantRequest = new MUCAdmin();
+            grantRequest.setTo(mucAddress);
+            grantRequest.setType(IQ.Type.set);
+            grantRequest.addItem(new MUCItem(MUCAffiliation.admin, conTwo.getUser().asBareJid()));
+
+            conOne.sendIqRequestAndWaitForResponse(grantRequest);
+
+            // Execute system under test.
+            final MUCAdmin revokeRequest = new MUCAdmin();
+            revokeRequest.setTo(mucAddress);
+            revokeRequest.setType(IQ.Type.set);
+            revokeRequest.addItem(new MUCItem(MUCAffiliation.none, conTwo.getUser().asBareJid()));
+
+            try {
+                conOne.sendIqRequestAndWaitForResponse(revokeRequest);
+
+                // Verify result.
+            } catch (XMPPException.XMPPErrorException e) {
+                fail("Expected owner '" + conOne.getUser() + "' to be able to revoke admin status from '" + conTwo.getUser().asBareJid() + "' (that is currently joined as '" + nicknameTarget+ "') in '" + mucAddress + "' (but the server returned an error).", e);
+            }
+        } finally {
+            // Tear down test fixture.
+            tryDestroy(mucAsSeenByOwner);
+        }
+    }
+
+    /**
+     * Verifies that an owner can revoke admin status from a user that is currently not in the room.
```

Comment:
Add disambiguation from the earlier test
```suggestion
     * Verifies that an owner can revoke admin status from a user that is currently not in the room, and give a reason.
```